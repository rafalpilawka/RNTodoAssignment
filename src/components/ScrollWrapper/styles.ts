import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';

export const styles = StyleSheet.create({
  contentContainer: {
    flexGrow: 1,
    paddingBottom: normalize(20),
    marginHorizontal: normalize(20),
    marginBottom: normalize(10),
  },
  scrollWrapper: {width: '100%'},
});
