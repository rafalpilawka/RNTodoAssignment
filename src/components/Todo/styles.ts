import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';

export const styles = StyleSheet.create({
  todo: {
    flexDirection: 'row',
    height: normalize(80),
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
  },
  text: {fontSize: normalize(16), flex: 1, paddingHorizontal: normalize(20)},
});
