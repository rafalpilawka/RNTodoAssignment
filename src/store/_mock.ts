export const mockTodos = [
  {id: 'aaa', title: 'Add those todos', completed: false},
  {id: 'aaa2233', title: 'Delete second', completed: false},
  {id: 'bbnjhjhjhjh', title: 'Add third one', completed: true},
];

export const mockLists = [
  {name: 'List 1', id: '11242342144'},
  {name: 'list 2', id: '11242342122'},
  {name: 'list 3', id: '1124234211'},
];

export const mockStore = {
  index: mockLists,
  todos: mockTodos,
  loading: false,
  errors: undefined,
  idLoader: undefined,
};

export const mockInitialStore = {
  index: [],
  todos: [],
  loading: false,
  errors: undefined,
  idLoader: undefined,
};
