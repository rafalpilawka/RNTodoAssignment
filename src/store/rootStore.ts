import {all, call} from 'redux-saga/effects';
import {applyMiddleware, combineReducers, createStore, Reducer} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import todosReducer from 'src/store/todos/todos.reducer';
import {watchTodosSaga} from 'src/store/todos/todos.saga';

const sagaMiddleware = createSagaMiddleware();

function* rootSaga() {
  yield all([
    call(watchTodosSaga),
    //OTHER WATCHERS GOES HERE
  ]);
}

// const store: Reducer<RootState> = combineReducers<RootState>({
//   todos: todosReducer,
// });

const rootStore = (state: any, action: any) => {
  return todosReducer(state, action);
};

// export interface RootState {
//   todos: ReduxState;
// }

export type RootState = ReturnType<typeof todosReducer>;

export default createStore(
  rootStore,
  composeWithDevTools(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSaga);
