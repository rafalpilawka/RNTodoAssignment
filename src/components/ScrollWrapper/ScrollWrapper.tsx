import React, {ReactElement, ReactNode} from 'react';
import {ScrollView, ViewStyle} from 'react-native';
import {styles} from './styles';

interface Props {
  children?: ReactNode;
  style?: ViewStyle;
  testId?: string;
}

export const ScrollWrapper = ({children, testId}: Props): ReactElement => {
  return (
    <ScrollView
      style={styles.scrollWrapper}
      testID={testId}
      keyboardShouldPersistTaps="always"
      bounces
      contentContainerStyle={styles.contentContainer}
      showsVerticalScrollIndicator={false}>
      {children}
    </ScrollView>
  );
};
