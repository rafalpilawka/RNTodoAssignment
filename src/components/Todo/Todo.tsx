import React, {ReactElement} from 'react';
import {Pressable, Text} from 'react-native';
import {RemoveButton} from 'src/components/RemoveButton/RemoveButton';
import {TodoType} from 'src/store/todos/todos.types';
import CheckBox from '@react-native-community/checkbox';
import {styles} from 'src/components/Todo/styles';

type PropsT = {
  item: TodoType;
  removeHandler: (id: string) => void;
  toggleHandler: (id: string) => void;
};

export const Todo = ({
  removeHandler,
  toggleHandler,
  item: {id, title, completed},
}: PropsT): ReactElement => {
  const _removeTodo = () => {
    removeHandler(id);
  };

  const _toggleCheckBox = () => {
    toggleHandler(id);
  };

  return (
    <Pressable style={styles.todo} onLongPress={() => console.log('longPress')}>
      <CheckBox value={completed} onValueChange={_toggleCheckBox} />
      <Text style={styles.text}>{title}</Text>
      <RemoveButton onPress={_removeTodo} />
    </Pressable>
  );
};
