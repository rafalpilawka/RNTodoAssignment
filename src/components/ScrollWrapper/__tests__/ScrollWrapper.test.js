import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {ScrollWrapper} from 'src/components/ScrollWrapper/ScrollWrapper';
import {Todo} from 'src/components/Todo/Todo';
import {mockTodos} from 'src/store/_mock';

const props = {
  children: mockTodos.map((el, idx) => (
    <Todo
      key={el.id}
      item={el}
      removeHandler={() => jest.fn()}
      toggleHandler={() => jest.fn()}
    />
  )),
  style: {backgroundColor: 'red'},
  testId: 'scrollView',
};

describe('ScrollWrapper', () => {
  let component;
  const text = 'Test';

  beforeEach(() => {
    component = shallow(<ScrollWrapper {...props} />);
    jest.clearAllMocks();
  });
  it('should match to snapshot', () => {
    const component = renderer.create(<ScrollWrapper {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render correct children', () => {
    const children = component.props().children;
    expect(children.length).toEqual(mockTodos.length);
  });
});
