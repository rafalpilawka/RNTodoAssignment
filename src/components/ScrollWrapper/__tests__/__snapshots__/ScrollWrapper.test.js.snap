// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollWrapper should match to snapshot 1`] = `
<RCTScrollView
  bounces={true}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "marginBottom": 20,
      "marginHorizontal": 40,
      "paddingBottom": 40,
    }
  }
  keyboardShouldPersistTaps="always"
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "width": "100%",
    }
  }
  testID="scrollView"
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 160,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={false}
        />
      </View>
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 32,
            "paddingHorizontal": 40,
          }
        }
      >
        Add those todos
      </Text>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 100,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          x
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 160,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={false}
        />
      </View>
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 32,
            "paddingHorizontal": 40,
          }
        }
      >
        Delete second
      </Text>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 100,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          x
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 160,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 32,
            "paddingHorizontal": 40,
          }
        }
      >
        Add third one
      </Text>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 100,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          x
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
